lib/google_drive.rb

```diff
@@ -122,34 +113,24 @@ def self.saved_session(path = nil, proxy = nil, client_id = nil, client_secret =
           :application_version => "0.4.0"
       )
       auth = client.authorization
-      auth.client_id = client_id
-      auth.client_secret = client_secret
-      auth.scope = [
-          "https://www.googleapis.com/auth/drive",
-          "https://spreadsheets.google.com/feeds/"
-      ]
-      auth.redirect_uri = "urn:ietf:wg:oauth:2.0:oob"
-
-      if token_data
+      auth.client_id = config.client_id
+      auth.client_secret = config.client_secret
+      auth.scope = config.scope.join(' ')
+      auth.redirect_uri = config.redirect_uris.first
 
-        auth.refresh_token = token_data["refresh_token"]
+      unless token_data.nil?
```

Comment:
I prefer "if token_data" here. Double-negation (unless ....nil?) is hard to read.